Apps/CesiumViewer/CesiumViewer.js

```diff
@@ -58,6 +60,12 @@ define([
             scene3DOnly : endUserOptions.scene3DOnly,
             requestRenderMode : true
         });
+
+        viewer.terrainProvider = new CesiumTerrainProvider({
```

Comment:
This causes the BaseLayerPicker to still think Ellipsoid is selected.  To fix that, you need to tell it to use terrain as configured by the picker.  Because there's a chance we might not have the baseLayer picker (because of the imagery provider options), we need to handle both cases.  Basically this code becomes:

```js
    var baseLayerPicker = viewer.baseLayerPicker;
    if (defined(baseLayerPicker)) {
        var viewModel = baseLayerPicker.viewModel;
        viewModel.selectedTerrain = viewModel.terrainProviderViewModels[1];
    } else {
        viewer.terrainProvider = new CesiumTerrainProvider({
            url: 'https://assets.agi.com/stk-terrain/v1/tilesets/world/tiles',
            requestWaterMask: true,
            requestVertexNormals: true
        });
    }
```

Reply:
As an aside, we can probably get rid of the imagery provider option, but for this PR I wouldn't worry about it.